### PR TITLE
Adds structure to hardware relationship, parse_hardware and missed attributes

### PR DESCRIPTION
**Pull request by Odravison**
Adding structure needed to relationship between physical server and hardware, such as:

1. `parse_hardware` method
2. `has_one` relationship into PhysicalServer model
3. adding some missed attributes into `parse_nodes` method

Depends on: https://github.com/ManageIQ/manageiq-providers-lenovo/pull/34

### CODE REVIEW

**Comment by Odravison** [comment]
@miq-bot add_labe WIP

**Comment by Odravison** [comment]
@miq-bot add_label WIP

**Comment by Odravison** [comment]
@miq-bot remove_label WIP

File: app/models/manageiq/providers/lenovo/physical_infra_manager/physical_server.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 module ManageIQ::Providers
   class Lenovo::PhysicalInfraManager::PhysicalServer < ::PhysicalServer
+    has_one :hardware
```
Comment (juliancheal) [inline app/models/manageiq/providers/lenovo/physical_infra_manager/physical_server.rb]: This isn't needed here, as this `Lenovo::PhysicalInfraManager::PhysicalServer` inherits from `PhysicalServer` in the main ManageIQ repo. That model has the dependencies. See this PR https://github.com/ManageIQ/manageiq/pull/14430

File: app/models/manageiq/providers/lenovo/physical_infra_manager/refresh_parser.rb [hunk 2]
```diff
@@ -69,23 +88,43 @@ def parse_firmware(firmware, uuid)
       }
     end
 
+    def parse_hardware(node)
```
Comment (juliancheal) [inline app/models/manageiq/providers/lenovo/physical_infra_manager/refresh_parser.rb]: This is great, I wasn't sure what data we needed to be parsed. 👍

Comment (Odravison) [inline app/models/manageiq/providers/lenovo/physical_infra_manager/refresh_parser.rb]: I know, i'm awesome.

Comment (walteraa) [inline app/models/manageiq/providers/lenovo/physical_infra_manager/refresh_parser.rb]: The data that we need parse are: total memory in `node.memoryModules`, total CPU cores and the maximum processor frequency in ` node.processors`, as you can see in the method. It was decided during a meeting with Lenovo guys.

**Comment by walteraa** [comment]
This PR should be closed.

File: app/models/manageiq/providers/lenovo/physical_infra_manager/refresh_parser.rb [hunk 3]
```diff
@@ -69,23 +88,43 @@ def parse_firmware(firmware, uuid)
...
-        :manufacturer  => node.manufacturer,
-        :machine_type  => node.machineType,
-        :model         => node.model,
-        :serial_number => node.serialNumber,
-        :uuid          => node.uuid,
-        :FRU           => node.FRU,
-        :macAddresses  => node.macAddress.split(",").flatten,
-        :ipv4Addresses => node.ipv4Addresses.split.flatten,
-        :ipv6Addresses => node.ipv6Addresses.split.flatten
+        :type                  => ManageIQ::Providers::Lenovo::PhysicalInfraManager::PhysicalServer.name,
```
Comment (juliancheal) [inline app/models/manageiq/providers/lenovo/physical_infra_manager/refresh_parser.rb]: This needs rebasing now that #34 has been merged.

Comment (walteraa) [inline app/models/manageiq/providers/lenovo/physical_infra_manager/refresh_parser.rb]: This PR should be closed and splited in two others:

1. Parse hardware informations(https://github.com/ManageIQ/manageiq-providers-lenovo/pull/39)
2. Add missing attributes

Comment (walteraa) [inline app/models/manageiq/providers/lenovo/physical_infra_manager/refresh_parser.rb]: The second one PR was opened: https://github.com/ManageIQ/manageiq-providers-lenovo/pull/40

**Comment by rodneyhbrown7** [comment]
Odravison can you close this PR?

**Comment by AndreyMenezes** [comment]
@juliancheal can you close this PR? Odravison don't work with us anymore.

**Pull request closed without merging**